avocado/utils/nfs.py

```diff
@@ -0,0 +1,349 @@
+        self.ori_exported = ori_exported
+        self.entry_tag = "%s_%s" % (self.path, self.client)
+        self.already_exported = False
+        self.ori_options = ""
+
+    def is_exported(self):
+        """
+        Check if the directory is already exported.
+
+        :return: If the entry is exported
+        :rtype: Boolean
+        """
+        ori_exported = self.ori_exported or nfs_exported()
+        if self.entry_tag in ori_exported.keys():
+            return True
+        return False
+
+    def need_reexport(self):
+        """
+        Check if the entry is already exported but the options are not
+        the same as we required.
+
+        :return: Need re export the entry or not
+        :rtype: Boolean
+        """
+        ori_exported = self.ori_exported or nfs_exported()
+        if self.is_exported():
+            exported_options = ori_exported[self.entry_tag]
+            options = [_ for _ in self.options if _ not in exported_options]
+            if options:
+                self.ori_options = exported_options
+                return True
+        return False
+
+    def unexport(self):
+        """
+        Unexport an entry.
+        """
+        if self.is_exported():
+            unexport_cmd = "exportfs -u %s:%s" % (self.client, self.path)
+            process.system(unexport_cmd)
+        else:
+            logging.warn("Target %s %s is not exported yet."
+                         "Can not unexport it." % (self.client, self.path))
+
+    def reset_export(self):
+        """
+        Reset the exportfs to the original status before we export the
+        specific entry.
+        """
+        self.unexport()
+        if self.ori_options:
+            tmp_options = self.options
+            self.options = self.ori_options.split(",")
+            self.export()
+            self.options = tmp_options
+
+    def export(self):
+        """
+        Export one directory if it is not in exported list.
+
+        :return: Export nfs file system succeed or not
+        """
+        if self.is_exported():
+            if self.need_reexport():
+                self.unexport()
+            else:
+                self.already_exported = True
+                logging.warn("Already exported target."
+                             " Don't need export it again")
+                return True
+        export_cmd = "exportfs"
+        if self.options:
+            export_cmd += " -o %s" % ",".join(self.options)
+        export_cmd += " %s:%s" % (self.client, self.path)
+        try:
+            process.system(export_cmd)
+        except process.CmdError, export_failed_err:
+            logging.error("Can not export target: %s" % export_failed_err)
+            return False
+        return True
+
+
+class Nfs(object):
+
+    """
+    Nfs class for handle nfs mount and umount. If a local nfs service is
+    required, it will configure a local nfs server accroding the params.
+    """
+
+    def __init__(self, params):
```

Comment:
missing docstring. It'd also be better to accept the basic params as arguments. If there are way too many arguments than it'd be acceptable to use `extra_args` or few `*_args` arguments related to certain set of features.

Briefly by looking at the meanings something like:

```python
def __init__(self, mount_dir, options, mount_src, setup_nfs=False, export_dir=None, export_ip=None, export_options=None)
```
but you might want to update it to fit the purpose (I'm a bit confused on which argument does what).